test_rmw_implementation/test/test_subscription.cpp

```diff
@@ -377,3 +377,52 @@ TEST_F(CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION), count_mismatched_subs
   EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
   EXPECT_EQ(0u, publisher_count);
 }
+
+TEST_F(
+  CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION),
+  take_serialized_with_info_with_bad_args) {
+  rmw_subscription_allocation_t * null_allocation{nullptr};  // still valid allocation
+  rcutils_allocator_t default_allocator = rcutils_get_default_allocator();
+  bool taken = false;
+  rmw_serialized_message_t serialized_message = rmw_get_zero_initialized_serialized_message();
+  rmw_message_info_t message_info = rmw_get_zero_initialized_message_info();
+  ASSERT_EQ(
+    RMW_RET_OK, rmw_serialized_message_init(
+      &serialized_message, 0lu, &default_allocator)) << rmw_get_error_string().str;
+
+  rmw_ret_t ret = rmw_take_serialized_message_with_info(
+    nullptr, &serialized_message, &taken,
+    &message_info, null_allocation);
+  EXPECT_EQ(RMW_RET_INVALID_ARGUMENT, ret) << rmw_get_error_string().str;
+  rmw_reset_error();
+
+  ret = rmw_take_serialized_message_with_info(
+    sub, nullptr, &taken, &message_info,
+    null_allocation);
```

Comment:
Minor suggestion, but I won't block PR on this
```suggestion
    sub, nullptr, &taken, &message_info, null_allocation);
```

Reply:
There are a couple like this, if you decide to change them, change them all

Reply:
Cpplint forced me to do them like that lol, it did not like  ```sub, nullptr, &taken, &message_info, null_allocation);```